### ISSUE TITLE
Support env interpolation in config keys

### ISSUE THREAD

**Issue by toolmantim** [buildkite/agent#408]
For the buildkite-plugins/docker-compose-buildkite-plugin#16 I tried to make the plugin test use the plugin itself, but I noticed the `$` interpolation wasn't happening.

```yaml
steps:
  - command: /hello
    plugins:
      "${BUILDKITE_BUILD_CHECKOUT_PATH}#${BUILDKITE_COMMIT}":
        run: helloworld
        config:
          - docker-compose.yml
          - docker-compose.test.yml
```

It seems we don't support `$` interpolation of env keys, only values.

**Comment by toolmantim** [comment]
The workaround is to use a dynamic pipeline, and let bash do the interpolation.

For example, `.buildkite/pipeline.sh | buildkite-agent pipeline upload`, where `.buildkite/pipeline.sh` is:

```bash
#!/bin/bash

set -eu

cat <<YAML
steps:
  - command: /hello
    plugins:
      "${BUILDKITE_BUILD_CHECKOUT_PATH}#${BUILDKITE_COMMIT}":
        run: helloworld
        config:
          - docker-compose.yml
          - docker-compose.test.yml
YAML
```

**Comment by lox** [comment]
Alternately, could you use a plugin name of `self` or something and have the agent resolve that to the above?

**Comment by lox** [comment]
Fixed in #578.

**Comment by toolmantim** [comment]
Noice @lox
